services/ansibler/server/nodeRequirments.go

```diff
@@ -13,35 +13,29 @@ const (
 	longhornReq = "../../ansible-playbooks/longhorn-req.yml"
 )
 
-func installLonghornRequirements(k8sNodepools []*NodepoolInfo) error {
+func installLonghornRequirements(nodepoolInfo *NodepoolInfo) error {
 	//since all nodes will be processed, fabricate dummy directory name with hash
-	idHash := utils.CreateHash(5)
-	directory := filepath.Join(baseDirectory, outputDirectory, idHash)
-	//generate private key files
-	for _, nodepoolInfo := range k8sNodepools {
-		if _, err := os.Stat(directory); os.IsNotExist(err) {
-			if err := os.MkdirAll(directory, os.ModePerm); err != nil {
-				return fmt.Errorf("failed to create directory %s : %w", directory, err)
-			}
-		}
-		if err := utils.CreateKeyFile(nodepoolInfo.PrivateKey, directory, fmt.Sprintf("%s.%s", nodepoolInfo.ID, privateKeyExt)); err != nil {
-			return fmt.Errorf("failed to create key file for %s : %w", nodepoolInfo.ID, err)
+	directory := filepath.Join(baseDirectory, outputDirectory, utils.CreateHash(utils.HashLength))
+
+	if _, err := os.Stat(directory); os.IsNotExist(err) {
```

Comment:
I see these functions used in many places, would it make sense to move them to `utils`, i.e. `utils.CreateDir()`?